aqt/installer.py

```diff
@@ -889,14 +960,12 @@ def run_list_qt_commercial(self, args) -> None:
         self.show_aqt_version()
 
         # Create temporary directory for installer
-        import shutil
-        from pathlib import Path
-
         temp_dir = Settings.qt_installer_temp_path
         temp_path = Path(temp_dir)
-        if temp_path.exists():
-            shutil.rmtree(temp_dir)
-        temp_path.mkdir(parents=True, exist_ok=True)
+        if not temp_path.exists():
+            temp_path.mkdir(parents=True, exist_ok=True)
+        else:
+            Settings.qt_installer_cleanup()
```

Comment:
Similar pattern appears 3 or 4 times in the code. I replaced the erasing of the aqt/tmp directory by the removal of its content instead. This make testing for the existence of files in the test suite much easier, as well as making the check for permissions easy by providing an existing folder in the file system of the user